src/io/linux.zig

```diff
@@ -29,6 +30,33 @@ pub const IO = struct {
     ios_queued: u64 = 0,
     ios_in_kernel: u64 = 0,
 
+    /// The head of a doubly-linked list of all operations that are:
+    /// - in the submission queue, or
+    /// - in the kernel, or
+    /// - in the completion queue, or
+    /// - in the `completed` list (excluding zero-duration timeouts)
+    awaiting: ?*Completion = null,
```

Comment:
Is it simpler to have awaiting track only the second point (in the kernel)? Meaning 1) on SQE acquire in `enqueue` then insert to list 2) on CQE recv in `flush_completions` then remove from list.